azure/durable_functions/models/utils/json_utils.py

```diff
@@ -36,6 +38,32 @@ def add_datetime_attrib(json_dict: Dict[str, Any], object_,
         json_dict[alt_name or attribute_name] = \
             getattr(object_, attribute_name).strftime(DATETIME_STRING_FORMAT)
 
+# When we recieve properties from WebJobs extension originally parsed as TimeSpan objects through Newtonsoft, 
+#   the format complies with the constant format specifier for TimeSpan in .NET. 
+#   See https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-timespan-format-strings#the-constant-c-format-specifier
+#   Python offers no convenient way to parse these back into timedeltas, so we use this regex method instead
+def parse_datetime_attrib_timespan(from_str: str) -> datetime.timedelta:
+    """Converts a string originally produced by TimeSpan.ToString("c") in .NET into python's timespan.timedelta
+
+    Parameters
+    ----------
+    from_str: The string format of the TimeSpan to convert
+
+    Returns
+    -------
+    timespan.timedelta
+        The TimeSpan expressed as a Python datetime.timedelta
+
+    """
+    match = re.match(r"^(-)?(?:([0-9]*)\.)?([0-9]{2}):([0-9]{2}):([0-9]{2})(?:\.([0-9]{7}))?$", from_str)
```

Comment:
nit -  as the meme goes, solving an issue with regex brings 2 new issues :-) .

Is this really necessary? I could have sworn our pre-existing Python parsing capabilities were able to parse dates just fine. Maybe that's not the case, just asking the obvious question out of due-dilligence.

Reply:
Dates (datetime) are easy to parse in Python. Timespans (timedelta) are not... This was the result of quite a bit of research and the best solution I could find

Reply:
Noted, I trust your research.

Maybe an improvement: I think there's a way to _name_ the regex capture groups, such as here: https://github.com/microsoft/dotnet-timespan-python/blob/5bb422e3f9ce37d116deb3a4fd335be5258c3e0d/src/package/timespan/timespan/timespan.py#L74-L82.

That might give the regex a bit more readability (thus debuggability)? But it's up to you.

Reply:
Great idea! Implemented it